plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoSession.java

```diff
@@ -141,16 +154,53 @@ public void testBuildQueryNestedField()
     @Test
     public void testProjectSufficientColumns()
     {
-        MongoColumnHandle col1 = new MongoColumnHandle("a", ImmutableList.of("b", "c"), BIGINT, false, false, Optional.empty());
-        MongoColumnHandle col2 = new MongoColumnHandle("a", ImmutableList.of("b"), BIGINT, false, false, Optional.empty());
-        MongoColumnHandle col3 = new MongoColumnHandle("a", ImmutableList.of(), BIGINT, false, false, Optional.empty());
-
-        List<MongoColumnHandle> output = getSufficientColumns(ImmutableList
-                .of(col1, col2, col3));
-        assertEquals(output.size(), 1);
-        assertEquals(output.get(0), col3);
-        output = getSufficientColumns(ImmutableList.of(col3, col2, col1));
-        assertEquals(output.size(), 1);
-        assertEquals(output.get(0), col3);
+        MongoColumnHandle col1 = createColumnHandle("a", BIGINT, ImmutableList.of("b", "d"));
```

Comment:
Can we have additional test coverage - where we have nested columns with the field names swapped - like a -> b -> c and a -> c -> b

Reply:
Is this what you looking for? Line 183-184

        MongoColumnHandle col5 = createColumnHandle("x", BIGINT, ImmutableList.of("a", "b", "c"));
        MongoColumnHandle col6 = createColumnHandle("x", BIGINT, ImmutableList.of("c", "a", "b"));

Reply:
Added